src/optimism/controller-v7.sol

```diff
@@ -0,0 +1,213 @@
+import "../interfaces/strategyv2.sol";
+import "../interfaces/onesplit.sol";
+import "../interfaces/converter.sol";
+import "../interfaces/univ3/IUniswapV3Pool.sol";
+
+contract ControllerV7 {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    address public constant burn = 0x000000000000000000000000000000000000dEaD;
+
+    address public governance;
+    address public strategist;
+    address public devfund;
+    address public treasury;
+    address public timelock;
+
+    mapping(address => address) public jars;
+    mapping(address => address) public strategies;
+    mapping(address => mapping(address => address)) public converters;
+    mapping(address => mapping(address => bool)) public approvedStrategies;
+    mapping(address => bool) public approvedJarConverters;
+
+    constructor(
+        address _governance,
+        address _strategist,
+        address _timelock,
+        address _devfund,
+        address _treasury
+    ) public {
+        governance = _governance;
+        strategist = _strategist;
+        timelock = _timelock;
+        devfund = _devfund;
+        treasury = _treasury;
+    }
+
+    function setDevFund(address _devfund) public {
+        require(msg.sender == governance, "!governance");
+        devfund = _devfund;
+    }
+
+    function setTreasury(address _treasury) public {
+        require(msg.sender == governance, "!governance");
+        treasury = _treasury;
+    }
+
+    function setStrategist(address _strategist) public {
+        require(msg.sender == governance, "!governance");
+        strategist = _strategist;
+    }
+
+    function setGovernance(address _governance) public {
+        require(msg.sender == governance, "!governance");
+        governance = _governance;
+    }
+
+    function setTimelock(address _timelock) public {
+        require(msg.sender == timelock, "!timelock");
+        timelock = _timelock;
+    }
+
+    function setJar(address _pool, address _jar) public {
+        require(msg.sender == strategist || msg.sender == governance, "!strategist");
+        jars[_pool] = _jar;
+    }
+
+    function approveStrategy(address _pool, address _strategy) public {
+        require(msg.sender == timelock, "!timelock");
+        approvedStrategies[_pool][_strategy] = true;
+    }
+
+    function revokeStrategy(address _pool, address _strategy) public {
+        require(msg.sender == governance, "!governance");
+        require(strategies[_pool] != _strategy, "cannot revoke active strategy");
+        approvedStrategies[_pool][_strategy] = false;
+    }
+
+    // in case of strategy stuck and if we need to relink the new strategy
+    function removeJar(address _pool) public {
+        require(msg.sender == strategist || msg.sender == governance, "!strategist");
+        jars[_pool] = address(0);
+    }
+
+    // in case of strategy stuck and if we need to relink the new strategy
+    function removeStrategy(address _pool) public {
```

Comment:
Good idea. I think something that might work equally well is a `setStrategyEmergency()`, which will naively set a new strategy without performing the `withdrawAll()` call